hatchery/nextflow.go

```diff
@@ -602,8 +612,63 @@ func ensureLaunchTemplate(ec2Svc *ec2.EC2, userName string, hostname string, job
 	}
 
 	if len(launchTemplate.LaunchTemplates) == 1 {
-		// TODO: Make sure user data in the existing launch template matches the user data
-		// we want (MIDRC-749)
+		// Check if the launch template has block device mappings
```

Comment:
is there a more generic way to compare the existing launch template to the one in the code? instead of only checking BlockDeviceMappings?

Also, this code wouldn't catch a difference in the BlockDeviceMappings definition, for example if we change the VolumeType, since we only check if there are BlockDeviceMappings at all

Maybe it's fine since this should be deprecated soon, but if we leave it as-is we should write a comment